### PR TITLE
Do not mark test methods as unused.

### PR DESCRIPTION
**Pull request by dbalek**
JUnit test methods should not be marked as unused in editor.

### CODE REVIEW

**Review by MartinBalin — APPROVED** [review]

File: java/java.hints/src/org/netbeans/modules/java/hints/bugs/Unused.java [hunk 1]
```diff
@@ -56,12 +62,33 @@ public class Unused {
     @TriggerTreeKind(Kind.COMPILATION_UNIT)
     public static List<ErrorDescription> unused(HintContext ctx) {
         List<UnusedDescription> unused = UnusedDetector.findUnused(ctx.getInfo(), () -> ctx.isCanceled());
+        Set<UnusedDescription> toRemove = new HashSet<>();
+        if (!unused.isEmpty()) {
+            List<TestMethodController.TestMethod> tms = getTestMethods(ctx.getInfo());
```
Comment (sdedic) [inline java/java.hints/src/org/netbeans/modules/java/hints/bugs/Unused.java:67]: I'd prefer some SPI that other modules could plug into. For example, `@RequestMapping` or other annotations could be put even on a private method (although discouraged), and should be deemed "used" -- tracked as #6284

**Review by sdedic — APPROVED** [review]

File: java/java.hints/src/org/netbeans/modules/java/hints/bugs/Unused.java (hunk 1 above)
Comment (dbalek) [reply java/java.hints/src/org/netbeans/modules/java/hints/bugs/Unused.java]: Required SPI created. Implementation changed to used the newly crated SPI.

**Review by lahodaj — APPROVED** [review]
Looks good to me, thanks!

**Comment by mbien** [comment]
@sdedic @dbalek does this PR close #6284?

**Pull request merged**